client-js/app/elements/labrad-registry.ts

```diff
@@ -387,25 +395,40 @@ export class LabradRegistry extends polymer.Base {
       try {
         this.$.pendingDialog.open();
         this.$.pendingOp.innerText = "Copying...";
-        var resp = await this.socket.copyDir({path: oldPath, dir: oldName, newPath: newPath, newDir: newName});
-        this.$.toastCopySuccess.show();
+        switch (this.$.dragClass.innerText) {
```

Comment:
I'm not a fan of passing data through the html text representation of the dialog element. Instead, I would suggest storing the data from the data transfer event in an attribute of the dialog, and then looking it up, e.g.:
```javascript
// in handleDrop
var data = JSON.parse(...);
this.$.dragDialog.dragData = data;

// here
switch (this.$.dragDialog.dragData['kind']) {
  ...
}
```

Of course, even better than this would be to have an alternate API for showing dialogs that returns a Promise which fires when the dialog is dismissed. Then we could just put the whole dialog flow inside a single async method and we wouldn't have to pass data through the DOM like this at all. But that's a change for another time.